tests/test_endpoint_buffers.py

```diff
@@ -93,3 +105,143 @@ def test_forward_results_are_generated(   # pylint: disable-msg=too-many-locals
                         assert (current_time - slot_length <=
                                 orders[0]["creation_time"] < current_time)
             current_time += slot_length
+
+    def test_prepare_results_for_publish(self, forward_setup):
+        area, _ = forward_setup
+        endpoint_buffer = SimulationEndpointBuffer(
+            job_id="JOB_1",
+            random_seed=41,
+            area=area,
+            should_export_plots=False)
+
+        output = endpoint_buffer.prepare_results_for_publish()
+
+        assert output == {
+            "job_id": "JOB_1",
+            "current_market": "",
+            "current_market_ui_time_slot_str": "",
+            "random_seed": 41,
+            "status": "",
+            "progress_info": {
+                "eta_seconds": 0,
+                "elapsed_time_seconds": 0,
+                "percentage_completed": 0,
+            },
+            "bids_offers_trades": {},
```

Comment:
Shouldn't we also write tests to check `bids_offers_trade` or `simulation_raw_data` attributes contain useful data?

Reply:
The reason why this is currently left out is that `bids_offers_trades` is populated using `self.flattened_area_core_stats_dict`. In turn, this is populated inside of `_populate_core_stats_and_sim_state()`. 
As I said somewhere, testing this method is tricky and takes quite long, because it has too many conditional branches and it's also overridden by the child `CoefficientEndpointBuffer` class. Testing all the `if` branches would create a lot of clutter in tests and make them difficult to understand and modify. That's why I think it would be better to refactor this method into a class first, and only then write tests for it.

Reply:
If you think it works, I might create a new ticket for this. Or we can keep this PR open and get back to it later on (since right now this is not the top priority because of EB tasks).

Reply:
I definitely agree. Let's have a new ticket for that. Thanks a lot for dealing with this.